src/main/java/org/veupathdb/service/eda/ds/plugin/correlationassaymetadata/CorrelationAssayMetadataBipartitenetworkPlugin.java

```diff
@@ -45,19 +50,46 @@ protected boolean computeGeneratesVars() {
   }
 
   @Override
-  protected void validateVisualizationSpec(EmptyDataPluginSpec pluginSpec) throws ValidationException {
+  protected void validateVisualizationSpec(CorrelationAssayMetadataBipartitenetworkSpec pluginSpec) throws ValidationException {
     // nothing to do here
   }
 
   @Override
-  protected List<StreamSpec> getRequestedStreams(EmptyDataPluginSpec pluginSpec) {
+  protected List<StreamSpec> getRequestedStreams(CorrelationAssayMetadataBipartitenetworkSpec pluginSpec) {
     // this plugin only uses the stats result of the compute; no tabular data streams needed
     return Collections.emptyList();
   }
 
   @Override
   protected void writeResults(OutputStream out, Map<String, InputStream> dataStreams) throws IOException {
-    System.out.println("I'm writing results!");
-    // writeComputeStatsResponseToOutput(out);
+
+    CorrelationAssayMetadataStatsResponse stats = getComputeResultStats(CorrelationAssayMetadataStatsResponse.class);
+
+
+    
+    useRConnectionWithRemoteFiles(Resources.RSERVE_URL, dataStreams, connection -> {
+      
+      // Take the stats response object and write it into R
+      // as a data.frame. There's probably a much more elegant solution, please advise!
+      connection.voidEval("statsDf <- data.frame(correlationCoef = character(), data1 = character(), data2 = character(), stringsAsFactors=FALSE)");
+      for (int i = 0; i < stats.getStatistics().toArray().length; i++) {
+        CorrelationPoint point = stats.getStatistics().get(i);
+        String newRow = "list(correlationCoef='" + point.getCorrelationCoef() + "', data1='" + point.getData1() + "', data2='" + point.getData2() + "')";
+        connection.voidEval("newRow <- " + newRow);
+        connection.voidEval("statsDf <- rbind(statsDf, newRow)");
+        
+      }
+
+      connection.voidEval("bpNet <- plot.data::bipartiteNetwork(" +
```

Comment:
i think usually wed have a wrapper sort of method that takes these params, builds the obj and writes to json as a single step. not essential, but might be nice for consistency.